packages/indy-sdk-to-askar-migration/src/IndySdkToAskarMigrationUpdater.ts

```diff
@@ -0,0 +1,363 @@
+import type { AnonCredsCredentialValue } from '@aries-framework/anoncreds'
+import type { Agent, FileSystem } from '@aries-framework/core'
+import type { EntryObject } from '@hyperledger/aries-askar-shared'
+
+import { AnonCredsCredentialRecord, AnonCredsLinkSecretRecord } from '@aries-framework/anoncreds'
+import { AskarWallet } from '@aries-framework/askar'
+import { InjectionSymbols, KeyDerivationMethod, JsonTransformer, TypedArrayEncoder } from '@aries-framework/core'
+import { Key, KeyAlgs, Store, StoreKeyMethod } from '@hyperledger/aries-askar-shared'
+
+import { IndySdkToAskarMigrationError } from './errors/IndySdkToAskarMigrationError'
+import { transformFromRecordTagValues } from './utils'
+
+/**
+ *
+ * Migration class to move a wallet form the indy-sdk structure to the new
+ * askar wallet structure.
+ *
+ * Right now, this is ONLY supported within React Native environments AND only sqlite.
+ *
+ * The reason it only works within React Native is that we ONLY update the
+ * keys, masterSecret and credentials for now. If you have an agent in Node.JS
+ * where it only contains these records, it may be used but we cannot
+ * guarantee a successful migration.
+ *
+ */
+export class IndySdkToAskarMigrationUpdater {
+  private store: Store
+  private walletId: string
+  private defaultLinkSecretId: string
+  private agent: Agent
+  private dbPath: string
+  private fs: FileSystem
+  private deleteOnFinish: boolean
+
+  private constructor(
+    store: Store,
+    walletId: string,
+    agent: Agent,
+    dbPath: string,
+    deleteOnFinish = false,
+    defaultLinkSecretId?: string
+  ) {
+    this.store = store
+    this.walletId = walletId
+    this.dbPath = dbPath
+    this.agent = agent
+    this.fs = this.agent.dependencyManager.resolve<FileSystem>(InjectionSymbols.FileSystem)
+    this.defaultLinkSecretId = defaultLinkSecretId ?? walletId
+    this.deleteOnFinish = deleteOnFinish
+  }
+
+  public static async initialize({
+    dbPath,
+    agent,
+    deleteOnFinish,
+    defaultLinkSecretId,
+  }: {
+    dbPath: string
+    agent: Agent
+    deleteOnFinish?: boolean
+    defaultLinkSecretId?: string
+  }) {
+    const {
+      config: { walletConfig },
+    } = agent
+    if (!walletConfig) throw new IndySdkToAskarMigrationError('Wallet config is required for updating the wallet')
```

Comment:
Can we also add a check if we're in Node.JS and throw an error?

Reply:
Yeah, would we do that with the instance type of the `dependencies` or something?

Reply:
```typeof process?.versions?.node !== 'undefined'```

Reply:
I just added this and we have tests in node.js so it would be good to keep support for Node.JS.

I could add a warning log that's it not fully working and a doc comment to the function? Or an override key that it can be used with caution?

Reply:
Yeah I think it can show a warning and also it will be documented so I don't see why it should be so strictly limited.